src/rust_api.rs

```diff
@@ -0,0 +1,1561 @@
+// Copyright 2016 6WIND S.A. <quentin.monnet@6wind.com>
+//
+// Licensed under the Apache License, Version 2.0 <http://www.apache.org/licenses/LICENSE-2.0> or
+// the MIT license <http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+//! Module provides API to create eBPF programs by Rust programming language
+
+use ebpf::Insn;
+
+/// Represents single eBPF instruction
+pub trait Instruction: Sized {
+    /// returns instruction opt code
+    fn opt_code_byte(&self) -> u8;
+
+    /// returns destination register
+    fn dst(&self) -> u8;
+
+    /// returns source register
+    fn src(&self) -> u8;
+
+    /// returns offset bytes
+    fn offset(&self) -> i16;
+
+    /// returns immediate value
+    fn imm(&self) -> i32;
+
+    /// sets destination register
+    fn dst_reg(self, dst: u8) -> Self;
+
+    /// sets source register
+    fn src_reg(self, src: u8) -> Self;
+
+    /// sets offset bytes
+    fn offset_bytes(self, offset: i16) -> Self;
+
+    /// sets immediate value
+    fn immediate(self, imm: i32) -> Self;
+}
```

Comment:
I see, that's too bad… I'm sure there is something we can do to refector this. But then it can be done later, when we think about something clean I guess.